src/NServiceBus.Core/Pipeline/Outgoing/InspectOutgoingMessageBehavior.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    class InspectOutgoingMessageBehavior : Behavior<IOutgoingPhysicalMessageContext>
```

Comment:
@andreasohlund as we discussed it. For me it fits fairly well into ```ImmediateDispatchTerminator```
It is wierd to add it to TransportOperations as I don't think it belong there.